llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6105,6 +6135,12 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
           if (PWSz == VL.size()) {
             ReuseShuffleIndicies.clear();
           } else {
+            if (UserTreeIdx.UserTE && UserTreeIdx.UserTE->isNonPowOf2Vec()) {
+              LLVM_DEBUG(dbgs() << "SLP: Reshuffling scalars not yet supported "
+                                   "for nodes with padding.\n");
+              newTreeEntry(VL, std::nullopt /*not vectorized*/, S, UserTreeIdx);
+              return false;
+            }
```

Comment:
Better to move this check and one below to line (or after) 6399 in the original source code to avoid duplication.

Reply:
I moved the check up the beginning of the `else { ` in the lambada to avoid the duplication. I think it cannot be moved out of `TryToFindDuplicates`, as we only need to have the bail out when reshuffling would be needed.